contracts/infrastructure/DappRegistry.sol

```diff
@@ -2,77 +2,197 @@ pragma solidity ^0.6.12;
+        require(registryOwners[_registryId] == address(0), "AR: duplicate registry");
+        registryOwners[_registryId] = _registryOwner;
+        emit RegistryCreated(_registryId, _registryOwner);
+    }
+
+    // Note: removeRegistry is not supported because that would allow the owner to replace registries that 
+    // have already been enabled by users with a new (potentially maliciously populated) registry 
+
+    function changeOwner(uint8 _registryId, address _newRegistryOwner) external onlyOwner(_registryId) {
+        require(_newRegistryOwner != address(0), "AR: new registry owner is 0");
+        registryOwners[_registryId] = _newRegistryOwner;
+        emit OwnerChanged(_registryId, _newRegistryOwner);
+    }
+
+    function requestTimelockChange(uint64 _newSecurityPeriod) external onlyOwner(0) {
+        newSecurityPeriod = _newSecurityPeriod;
+        securityPeriodChangeAfter = uint64(block.timestamp) + securityPeriod;
+        emit TimelockChangeRequested(_newSecurityPeriod);
     }
 
-    // Do we want to let the owner to delete a registry?
-    function createRegistry(bytes32 _registry, address _manager) external onlyOwner {
-        require(_registry != bytes32(0) && _manager != address(0), "AR: invalid parameters");
-        require(registryManager[_registry] == address(0), "AR: duplicate registry");
-        registries.push(_registry);
-        registryManager[_registry] = _manager;
-        emit RegistryCreated(_registry, _manager);
+    function confirmTimelockChange() external {
+        uint64 newPeriod = newSecurityPeriod;
+        require(securityPeriodChangeAfter > 0 && securityPeriodChangeAfter <= block.timestamp, "AR: can't (yet) change timelock");
+        securityPeriod = newPeriod;
+        newSecurityPeriod = 0;
+        securityPeriodChangeAfter = 0;
+        emit TimelockChanged(newPeriod);
     }
 
-    // need to add timelock
-    function addAuthorisationToRegistry(bytes32 _registry, address _contract, address _filter) external {
-        if (_registry == ARGENT_DEFAULT) {
-            require(msg.sender == owner, "AR: not authorised");
+    /**************  Management of registries' content  *****************/
+
+    /**
+    * @notice Set an authorisation filter for a dapp that has _not_ previously been authorised or 
+    * that has previously been authorised _without_ a filter. To change a non-zero filter, use `requestFilterUpdate`
+    * and `confirmFilterUpdate` instead.
+    * @param _registryId The id of the registry to modify
+    * @param _dapp The address of the dapp contract to authorise.
+    * @param _filter The address of the filter contract to use.
+    */
+    function addFilter(uint8 _registryId, address _dapp, address _filter) external onlyOwner(_registryId) {
+        uint auth = uint(authorisations[_registryId][_dapp]); // {filter:160}{validFrom:64}
+        require((auth >> 64) == 0, "DR: filter already set");
+        uint validFrom = auth & 0xffffffffffffffff;
```

Comment:
Just for clarity can we rename `validFrom` to `validAfter`? One is inclusive of the timestamp and the other is not, and that's the name used in declaration notes and other functions here.